format/index.md

```diff
@@ -52,761 +58,508 @@ called out below.
-by their type in a document.
+Every JSON API resource object is uniquely identified by a combination of
+its `type` and `id` members. This identification is used for linkage in
+compound documents and batch operations that modify multiple resources at a
+time.
 
-Each resource object **MAY** contain a `"type"` key to explicitly designate its
-type.
-
-The `"type"` key is **REQUIRED** when the type of a resource is not otherwise
-specified in a document.
-
-#### Resource URLs <a href="#document-structure-resource-urls" id="document-structure-resource-urls" class="headerlink"></a>
+A resource object's `type` and `id` pair **MUST** refer to a single, unique
+resource.
 
-The URL of each resource object **MAY** be specified with the `"href"` key.
-Resource URLs **SHOULD** only be specified by the server and therefore are
-typically only included in response documents.
-
-```javascript
-//...
-  [{
-    "id": "1",
-    "href": "http://example.com/comments/1",
-    "body": "Mmmmmakase"
-  }, {
-    "id": "2",
-    "href": "http://example.com/comments/2",
-    "body": "I prefer unagi"
-  }]
-//...
-```
-
-A server **MUST** respond to a `GET` request to the specified URL with a
-response that includes the resource.
-
-It is generally more efficient to specify URL templates at the root level of a
-response document rather than to specify individual URLs per resource.
-
-#### Resource Relationships <a href="#document-structure-resource-relationships" id="document-structure-resource-relationships" class="headerlink"></a>
-
-The value of the `"links"` key is a JSON object that represents linked
-resources, keyed by the name of each association.
-
-For example, the following post is associated with a single `author` and a
-collection of `comments`:
-
-```javascript
-//...
-  {
-    "id": "1",
-    "title": "Rails is Omakase",
-    "links": {
-      "author": "9",
-      "comments": [ "5", "12", "17", "20" ]
-    }
-  }
-//...
-```
-
-##### To-One Relationships <a href="#document-structure-resource-relationships-to-one" id="document-structure-resource-relationships-to-one" class="headerlink"></a>
-
-To-one relationships **MUST** be represented with one of the formats for
-individual resources described above.
+#### Resource Types <a href="#document-structure-resource-types" id="document-structure-resource-types" class="headerlink"></a>
 
-For example, the following post is associated with a single author, identified
-by ID:
+Each resource object **MUST** contain a `type` member, whose value **MUST**
+be a string. The `type` is used to describe resource objects that share
+common attributes and relationships.
 
-```javascript
-//...
-  {
-    "id": "1",
-    "title": "Rails is Omakase",
-    "links": {
-      "author": "17"
-    }
-  }
-//...
-```
+> Note: This spec is agnostic about inflection rules, so `type` can be either
+plural or singular. However, the same value should be used consistently
```

Comment:
Just to be clear (and I'm probably mis-reading your comment), we're not discussing the key `type` vs `types`, but rather the value associated with the `type` member (e.g. `comment` vs `comments`, `blog-post` vs `blog-posts`). So the challenge becomes not only making rules about pluralization but also word delimiters. All together it seems consistent to leave naming decisions, including those for types, attributes, associations and URLs, as recommendations rather than requirements. If any of these can be freely named, then why not all of them?